packages/passwordless-auth-construct/src/sign-up/construct.ts

```diff
@@ -0,0 +1,103 @@
+import { Construct } from 'constructs';
+import {
+  NodejsFunction,
+  NodejsFunctionProps,
+  OutputFormat,
+} from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+import { Architecture, Runtime } from 'aws-cdk-lib/aws-lambda';
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import { IUserPool } from 'aws-cdk-lib/aws-cognito';
+import { Effect, Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { CfnOutput } from 'aws-cdk-lib';
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+/**
+ * Amplify SignUp Passwordless construct
+ */
+export class AmplifySignUpPasswordless extends Construct {
+  /**
+   * Create a new SignUp Passwordless construct
+   * verifyAuthChallengeResponse function is needed in order to attach policy
+   * userPool is needed to identity the resource and create the policy for verifyAuthChallengeResponse
+   */
+  constructor(
+    scope: Construct,
+    id: string,
+    verifyAuthChallengeResponse: NodejsFunction,
+    userPool: IUserPool
+  ) {
+    super(scope, id);
+
+    // default memory allocation for lambda functions
+    const defaultMemorySize = 128;
+
+    const commonOptions: NodejsFunctionProps = {
+      entry: path.join(dirname, '.', 'create_user_handler.js'),
+      runtime: Runtime.NODEJS_20_X,
+      architecture: Architecture.ARM_64,
+      bundling: {
+        format: OutputFormat.ESM,
+      },
+    };
+
+    const createUserFunction = new NodejsFunction(scope, `CreateUser${id}`, {
+      handler: 'createUser',
+      ...commonOptions,
+      memorySize: defaultMemorySize,
+    });
+
+    const api = new apigateway.RestApi(this, `CreateUserApi${id}`, {
+      restApiName: 'Create User service',
+      description: 'This service creates users.',
+      defaultCorsPreflightOptions: {
+        allowOrigins: apigateway.Cors.ALL_ORIGINS,
+      },
+    });
+
+    const putCreateUserIntegration = new apigateway.LambdaIntegration(
+      createUserFunction,
+      {
+        requestTemplates: { 'application/json': '{ "statusCode": "200" }' },
+      }
+    );
+
+    api.root.addMethod('PUT', putCreateUserIntegration);
+
+    new CfnOutput(this, 'apiUrl', { value: api.url });
```

Comment:
if this URL need to make it's way back to the client config file, we'll need to use [`addBackendOutputEntry`](https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts#L25). You can see how this is used in the data construct [here](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts#L262)

We could also mark this as a TODO for later

Reply:
Please also create GH issue for this and link in the comment. This should be release blocker.